classes/affiliation/Collector.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * @file classes/affiliation/Collector.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Collector
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of affiliations
+ */
+
+namespace PKP\affiliation;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\Hook;
+
+/**
+ * @template T of Affiliation
+ */
+class Collector implements CollectorInterface
+{
+    public DAO $dao;
+
+    public ?int $count = null;
+
+    public ?int $offset = null;
+
+    /** @var int|null Get affiliations with author id */
+    public ?int $authorId = null;
+
+    /** @var int[]|null Get affiliations with author ids */
+    public ?array $authorIds = null;
+
+    /** Get affiliations with a name */
+    public ?string $name = null;
+
+    /** Get affiliations with a searchPhrase */
+    public ?string $searchPhrase = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /** @copydoc DAO::getCount() */
+    public function getCount(): int
+    {
+        return $this->dao->getCount($this);
+    }
+
+    /** @copydoc DAO::getIds() */
+    public function getIds(): Collection
+    {
+        return $this->dao->getIds($this);
+    }
+
+    /** @copydoc DAO::getMany() */
+    public function getMany(?string $submissionLocale = null): LazyCollection
+    {
+        return $this->dao->getMany($this, $submissionLocale);
+    }
+
+    /**
+     * Filter by single author
+     */
+    public function filterByAuthorId(?int $authorId): self
+    {
+        $this->authorId = $authorId;
+        return $this;
+    }
+
+    /**
+     * Filter by authors
+     */
+    public function filterByAuthorIds(?array $authorIds): self
+    {
+        $this->authorIds = $authorIds;
+        return $this;
+    }
```

Comment:
It seems this function is never used? 
Then maybe to leave the filtering only by one author ID?